plugin/service/host/state.go

```diff
@@ -86,16 +73,68 @@ func (s *awsCatalogPersistedState) toProto() (*pb.HostCatalogPersisted, error) {
 
 // EC2Client returns a configured EC2 client based on the session
 // information stored in the state.
-func (s *awsCatalogPersistedState) EC2Client(ctx context.Context) (EC2API, error) {
+func (s *awsCatalogPersistedState) EC2Client(ctx context.Context, opt ...ec2Option) (EC2API, error) {
+	opts, err := getOpts(opt...)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing options when fetching EC2 client: %w", err)
+	}
 	awsCfg, err := s.GenerateCredentialChain(ctx)
 	if err != nil {
 		return nil, fmt.Errorf("error getting AWS configuration when fetching EC2 client: %w", err)
 	}
 	if awsCfg == nil {
 		return nil, fmt.Errorf("nil aws configuration")
 	}
+	var ec2Opts []func(*ec2.Options)
+	ec2Opts = append(ec2Opts, ec2.WithEndpointResolverV2(&endpointResolver{
```

Comment:
If dualStack is set to false, do we still need to pass in endpointResolver? Could this be conditional?

Reply:
Yea, that makes sense I will change it.